### PR TITLE
6398 Improve check and error message of bundle run

### LINKED ISSUE #6398: Improve arg check and error for `bundle run`
**Is your feature request related to a problem? Please describe.**
Ticket reported by internal QA team:
MONAI 1.1 train --
python -m monai.bundle run train --config_file configs/train.json
MONAI 1.2 train --
python -m monai.bundle run --config_file configs/train.json

If user happen to write old API in new MONAI, no information pointing out the possible root cause. 
The job ends without doing any thing, and no error message:
```
04/18/2023 06:16:29 AM bundle_test _run_commands INFO: Executing export PYTHONPATH=$PYTHONPATH:/workspace/bundles/monai_pathology_nuclei_segmentation_classification_v0.1.7 && echo $PYTHONPATH && python -m monai.bundle run training --meta_file /workspace/bundles/monai_pathology_nuclei_segmentation_classification_v0.1.7/configs/metadata.json --config_file /workspace/bundles/monai_pathology_nuclei_segmentation_classification_v0.1.7/configs/train.json --logging_file /workspace/bundles/monai_pathology_nuclei_segmentation_classification_v0.1.7/configs/logging.conf --network_def#freeze_encoder false --network_def#pretrained_url None --stage 1 2>&1 | tee running.log
:/workspace/bundles/monai_pathology_nuclei_segmentation_classification_v0.1.7
2023-04-18 06:16:33,631 - INFO - --- input summary of monai.bundle.scripts.run ---
2023-04-18 06:16:33,631 - INFO - > run_id: 'training'
2023-04-18 06:16:33,632 - INFO - > meta_file: '/workspace/bundles/monai_pathology_nuclei_segmentation_classification_v0.1.7/configs/metadata.json'
2023-04-18 06:16:33,632 - INFO - > config_file: '/workspace/bundles/monai_pathology_nuclei_segmentation_classification_v0.1.7/configs/train.json'
2023-04-18 06:16:33,632 - INFO - > logging_file: '/workspace/bundles/monai_pathology_nuclei_segmentation_classification_v0.1.7/configs/logging.conf'
2023-04-18 06:16:33,632 - INFO - > network_def#freeze_encoder: 'false'
2023-04-18 06:16:33,632 - INFO - > stage: 1
2023-04-18 06:16:33,632 - INFO - ---


2023-04-18 06:16:33,632 - INFO - Setting logging properties based on config: /workspace/bundles/monai_pathology_nuclei_segmentation_classification_v0.1.7/configs/logging.conf.
```

Please consider adding some hints (in the script or doc) to reduce the confusion during the API migration.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #6398 .

### Description

This PR added check and error message for `bundle run` API.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/bundle/scripts.py [hunk 1]
```diff
@@ -625,8 +625,11 @@ def run(
 
     Args:
         run_id: ID name of the expected config expression to run, default to "run".
+            to run the config, the target config must contain this ID.
         init_id: ID name of the expected config expression to initialize before running, default to "initialize".
+            allow a config to have no `initialize` logic and the ID.
```
Comment (wyli) [inline monai/bundle/scripts.py:630]:
```suggestion
            It's optional for both configs and `run`.
```

**Review by wyli — APPROVED** [review]

File: monai/bundle/scripts.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/bundle/scripts.py]: Sure, updated in the latest commit, I didn't change the doc-string in the `ConfigWorkFlow`, the logic is slightly different.

Thanks.

**Pull request merged**